File: Actly.API/Models/Event.cs
using System.ComponentModel.DataAnnotations;

namespace Actly.API.Models
{
    public class Event
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public required string Title { get; set; }

        public int? OrganizerId { get; set; }

        public User? Organizer { get; set; } = null!;

        [Required]
        public required string Description { get; set; }

        [Required]
        public DateTime Date { get; set; }

        [Required]
        public required string Location { get; set; }

        [Required]
        public int MaxParticipants { get; set; }

        [Required]
        public required string Category { get; set; }

        public required List<Participation> Participations { get; set; }
    }
}

File: Actly.Client/src/app/models/event.ts
export interface Event {
  id: number;
  title: string;
  organizerId?: number | null;
  description: string;
  date: Date;
  location: string;
  maxParticipants: number;
  category: string;
  attended: boolean;
  participants: Participant[];
}

export interface Participant {
  id: number;
  attended: boolean;
  username: string;
  userId: number;
}
